src/smart-escrow/SmartEscrow.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "./Errors.sol";
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+/**
+ * @title SmartEscrow contract
+ * @dev Contract to handle payment of OP tokens over a period of vesting with
+ * the ability to terminate the contract.
+ */
+contract SmartEscrow is Ownable2Step {
+    event OPTransfered(uint256 amount, address indexed recipient);
+    event BeneficiaryOwnerUpdated(address indexed oldOwner, address indexed newOwner);
+    event BeneficiaryUpdated(address indexed oldBeneficiary, address indexed newBeneficiary);
+    event ContractTerminated();
+
+    IERC20 public constant OP_TOKEN = IERC20(0x4200000000000000000000000000000000000042);
+
+    address public beneficiaryOwner;
+    address public beneficiary;
+    uint256 public released;
+    uint64 public immutable start;
+    uint64 public immutable end;
+    uint64 public immutable vestingPeriod;
+    uint64 private immutable _initialTokens;
+    uint64 private immutable  _vestingEventTokens;
```

Comment:
thoughts on private vs public for these?

Reply:
mm no strong opinion, also I think we can just use uint256 for everything for simplicity? It only saves you a little gas on deploy